fm2prof/RegionPolygonFile.py

```diff
@@ -186,36 +188,57 @@ def _check_overlap(self):
                             level="warning",
                         )
 
+    @property
+    def polygons(self) -> list[Polygon]:
+        return self._polygons
+
+    @polygons.setter
+    def polygons(self, polygons_list: List[Polygon]) -> None:
+        if not all([isinstance(polygon, Polygon) for polygon in polygons_list]):
+            raise ValueError("Polygons must be of type Polygon")
+        # Check if properties contain the required 'name' property
+        names = [polygon.properties.get("name") for polygon in polygons_list]
+        if not all(names):
+            raise ValueError("Polygon properties must contain key-word 'name'")
+        # Check if 'name' property is unique, otherwise _check_overlap will produce bugs
+        if len(names) != len(set(names)):
+            raise ValueError("Property 'name' must be unique")
+        self._polygons = polygons_list
+
 
 class RegionPolygonFile(PolygonFile):
     def __init__(self, region_file_path, logger):
         super().__init__(logger)
         self.read_region_file(region_file_path)
 
     @property
-    def regions(self):
+    def regions(self) -> list[Polygon]:

```

Comment:
Hmm..So `regions` is the same as `self.polygons` which is simply `self._polygons`? What added value does this property have? Why not use `self.polygons` 🤔